api/src/play_asset_pack_request_tracker.gd

```diff
@@ -0,0 +1,70 @@
+# ##############################################################################
+#
+#	Copyright 2020 Google LLC
+#
+#	Licensed under the Apache License, Version 2.0 (the "License");
+#	you may not use this file except in compliance with the License.
+#	You may obtain a copy of the License at
+#
+#		https://www.apache.org/licenses/LICENSE-2.0
+#
+#	Unless required by applicable law or agreed to in writing, software
+#	distributed under the License is distributed on an "AS IS" BASIS,
+#	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#	See the License for the specific language governing permissions and
+#	limitations under the License.
+#
+# ##############################################################################
+#
+# The PlayAssetPackRequestTracker class generates unique signal_id integers and
+# keeps a mapping of signal_id to Request objects. This signal_id is used as an
+# identifier and passed to the Android plugin. Eventually the Android plugin will
+# emit signals along with this signal_id. In this way we can know which signal 
+# emitted from the plugin corresponds to which Request object
+#
+# ##############################################################################
+class_name PlayAssetPackRequestTracker
+extends Object
+
+const _SIGNAL_ID_MAX = 9223372036854775807
+const _SIGNAL_ID_MIN = 0
+
+var _signal_id_counter : int
+var _signal_id_to_request_map : Dictionary
+
+func _init():
+	# On first request's registration, _signal_id_counter should be incremented to _SIGNAL_ID_MIN
+	_signal_id_counter = _SIGNAL_ID_MIN - 1
```

Comment:
I think its fine to initialize this to `_SIGNAL_ID_MIN` to simplify the code and make it so readers don't need to know how GDScript handles underflow to know what you intend here.